pos_partner_firstname/static/src/js/screens.js

```diff
@@ -0,0 +1,94 @@
+odoo.define("pos_partner_firstname.screens", function (require) {
+    "use strict";
+    const {parse} = require("web.field_utils");
+    const PosComponent = require("point_of_sale.PosComponent");
+    const Registries = require("point_of_sale.Registries");
+    const ClientDetailsEdit = require("point_of_sale.ClientDetailsEdit");
+
+    const ClientDetailsEditFirstname = (ClientDetailsEdit) =>
+        class extends ClientDetailsEdit {
+            constructor() {
+                super(...arguments);
+                var self = this;
+                this.rpc({
+                    model: "res.partner",
+                    method: "get_names_order",
+                    args: [],
+                }).then(function (partner_names_order) {
+                    if (partner_names_order != false) {
+                        self.partner_names_order = "last_first";
+                    }
+                });
+            }
+
+            mounted() {
+                this.display_client_details(this.props.partner);
+                super.mounted(...arguments);
+            }
+            willUnmount() {
+                this.display_client_details(this.props.partner);
+                super.willUnmount(...arguments);
+            }
+
+            _update_client_name(checked) {
+                if (!checked) {
+                    var lastname = $(".lastname").val() || "";
+                    var firstname = $(".firstname").val() || "";
+                    var name = null;
+                    if (this.partner_names_order === "last_first_comma") {
+                        name = lastname + ", " + firstname;
+                    } else if (this.partner_names_order === "first_last") {
+                        name = firstname + " " + lastname;
+                    } else {
+                        name = lastname + " " + firstname;
+                    }
+                    $(".client-name").val(name);
+                }
+            }
+
+            updatePerson() {
+                this.display_client_details(this.props.partner);
+            }
+
+            display_client_details(partner) {
```

Comment:
You shouldn't need to do all this in plain js/jquery here.

You should use `owl` and `qweb` attributes to achieve this, look for example [the `t-on-click` or `t-attf-class` on this line](https://github.com/odoo/odoo/blob/77287edb77fb53836af2aa70c7b14c24e23d6002/addons/point_of_sale/static/src/xml/Screens/ProductScreen/Orderline.xml#L5), should give you a pointer on how to achieve it

Reply:
@ivantodorovich thank you for the suggestions!